src/main/java/hudson/plugins/tfs/model/HttpMethod.java

```diff
@@ -0,0 +1,61 @@
+package hudson.plugins.tfs.model;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.ProtocolException;
+import java.nio.charset.Charset;
+
+public enum HttpMethod {
+    GET {
+        @Override
+        public void sendRequest(final HttpURLConnection connection, final String body) throws IOException{
+            // do nothing for GET
+        }
+    },
+    POST,
+    HEAD {
+        @Override
+        public void sendRequest(final HttpURLConnection connection, final String body) throws IOException{
+            // do nothing for HEAD
+        }
+    },
+    PATCH,
+    OPTIONS,
+    PUT,
+    DELETE,
+    TRACE,
+    ;
+
+    private static final Charset UTF8 = Charset.forName("UTF-8");

```

Comment:
That's a good idea.  `StandardCharsets` is Java 1.7+ and we're currently targeting Java 1.6, so I'll add an item in the backlog to upgrade to Java 1.7 (as most of Jenkins has) and to make that change at the same time.